pkg/controllers/deleting/controller.go

```diff
@@ -141,7 +154,11 @@ func (c *controller) deleting(ctx context.Context, logger logr.Logger, ePolicy e
 	for _, kind := range kinds {
 		debug := debug.WithValues("kind", kind)
 		debug.Info("processing...")
-		list, err := c.client.ListResource(ctx, "", kind, "", spec.MatchConstraints.ObjectSelector)
+		listNamespace := policyNamespace
+		if listNamespace != "" && !isNamespacedKind(kind, restMapper) {
+			listNamespace = ""
```

Comment:
does this mean that a NamespacedDeletingPolicy is able to delete a Cluster scoped Resource?

I think this should not be possible. A namespaced scope policy should only be able to delete resources in this namespace.

cc @eddycharly